### PR TITLE
Show/Hide API Password

### LINKED ISSUE #206: password in plugin api settings displayed in plain text by default
**Describe the bug**
Every time you show someone the plugin you end up leaking your API credentials because they are written in plain text in the API settings tab that you first see.

**To Reproduce**
1. Go to 'Plugin settings'
2. Click on nothing
3. See your api password in plain text

**Expected behavior**
the api password should be hidden by default.
Optionally it should be displayed only when you click on an 'eye' symbol

**Environment (please complete the following information):**
 - Plugin version: since forever

This could also be seen as an enhancement as well as a bug therefore it has both tags :)

### PR DESCRIPTION
**Pull request by ferishili**
This PR fixes #206,

### Description
refer to the issue

### How it works
- The API password now gets the dot mask when the page is loaded
- The ability to show/hide the password is also provided, via the eye icon on the right side of the password input.
- (NEW) The xoctPasswordMask js module is now introduced, which makes it possible to mask multiple inputs at the same time, by passing the json string array of input ids
- (NEW) The xoctPasswordMask js can be used in different places, not limited only to curl_password!

### Important to note:
- 2 svg files have been added into /templates/images
- 1 js file and 1 css file responsible to do the job are also added

### CODE REVIEW

**Review by reiferschris — APPROVED** [review]
Works as advertised :D Thank you. 
Finally, no more spoiling the api password when showing someone the plugin settings.

File: classes/Conf/class.xoctConfFormGUI.php [hunk 1]
```diff
@@ -159,6 +166,11 @@ public static function checkItem($item)
      */
     protected function initAPISection()
     {
+        global $DIC;
```
Comment (chfsx) [inline classes/Conf/class.xoctConfFormGUI.php:169]: @ferishili could you move this (as $this->main_template = ...) to the constructor?

File: classes/Conf/class.xoctConfFormGUI.php [hunk 2]
```diff
@@ -159,6 +166,11 @@ public static function checkItem($item)
...
+        $main_tpl = $DIC->ui()->mainTemplate();
+        $main_tpl->addCss($this->plugin->getStyleSheetLocation('default/password_toggle.css'));
+        $main_tpl->addJavaScript($this->plugin->getStyleSheetLocation('default/password_toggle.min.js'));
```
Comment (chfsx) [inline classes/Conf/class.xoctConfFormGUI.php:172]: this should be changes to ->addOnloadCode() and you should pass something like an `init('#elemt_id')` method. please have a look to my comment in password_toggle.js

File: templates/default/password_toggle.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+$(document).ready(function() {
```
Comment (chfsx) [inline templates/default/password_toggle.js:1]: Please rewrite this according to https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/docs/development/js-bundling.md

this results in a small module which then can be used with an init-method (see comment above) like PasswordToggle.init().

I think that we should increasingly centralize JS in the plugin and gradually bring it into this form. we can therefore start directly with this JS

File: templates/default/password_toggle.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	let password_input = document.getElementById('curl_password');
```
Comment (chfsx) [inline templates/default/password_toggle.js:2]: the modification also means that we no longer have a direct bundling to exactly this element but the functionality could also be used for other password inputs.

**Review by chfsx — CHANGES_REQUESTED** [review]
Hi! thanks for this suggestion, I ahbe some change requests written as comments.

but basically the following question arises: there is a lot of new code for this functionality. as soon as we rebuild the forms (from legacy-forms to UI-components) we will have to rework all this again. but if you want to do this at the moment, i can't decide, maybe @Amstutz could give an opinion.

otherwise the PR would be ok after the changes, best thanks for the work!

**Comment by chfsx** [comment]
HI @ferishili I opened a PR with the changes to move the JS code to the new structure: https://github.com/elan-ev/ilias-opencast-plugin/pull/3

**Comment by ferishili** [comment]
Hi @chfsx, thanks a lot.

**Review by chfsx — APPROVED** [review]
Thanks a lot for the changes!

**Comment by Amstutz** [comment]
Thx a lot!

**Pull request merged**